src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -33,56 +33,113 @@ public PropertySubpatternCompletionProvider()
-
-            // Filter out those members that have already been typed
-            var propertyPatternClause = (PropertyPatternClauseSyntax)token.Parent;
+            var members = GetCandidatePropertiesAndFields(document, semanticModel, position, type);
 
-            // List the members that are already tested in this property sub-pattern
-            var alreadyTestedMembers = new HashSet<string>(propertyPatternClause.Subpatterns.Select(
-                // PROTOTYPE(extended-property-patterns) ExpressionColon
-                p => p.NameColon?.Name.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
+            if (propertyPatternClause is not null)
+            {
+                // Filter out those members that have already been typed as simple (not extended) properties
+                var alreadyTestedMembers = new HashSet<string>(propertyPatternClause.Subpatterns.Select(
+                    p => p.NameColon?.Name.Identifier.ValueText).Where(s => !string.IsNullOrEmpty(s)));
 
-            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name) &&
-                m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+                members = members.Where(m => !alreadyTestedMembers.Contains(m.Name));
+            }
 
-            foreach (var untestedMember in untestedMembers)
+            foreach (var member in members)
             {
                 const string ColonString = ":";
                 context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
-                    displayText: untestedMember.Name.EscapeIdentifier(),
+                    displayText: member.Name.EscapeIdentifier(),
                     displayTextSuffix: ColonString,
                     insertionText: null,
-                    symbols: ImmutableArray.Create(untestedMember),
-                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    symbols: ImmutableArray.Create(member),
+                    contextPosition: context.Position,
                     rules: s_rules));
             }
+
+            return;
+
+            static ITypeSymbol GetMemberAccessType(ITypeSymbol type, ExpressionSyntax expression, Document document, SemanticModel semanticModel, int position)
+            {
+                string name;
+                if (expression is MemberAccessExpressionSyntax memberAccess)
+                {
+                    type = GetMemberAccessType(type, memberAccess.Expression, document, semanticModel, position);
+                    name = memberAccess.Name.Identifier.ValueText;
+                }
+                else if (expression is IdentifierNameSyntax identifier)
+                {
+                    name = identifier.Identifier.ValueText;
+                }
+                else
+                {
+                    return null;
+                }
+
+                return GetMemberType(type, name, document, semanticModel, position);
```

Comment:
i'm pretty confused with teh data flow here :)  any way to make thsi simpler by just separating out the 'is dotted name' and 'is identifer name' cases?